server/src/main/java/org/opensearch/index/codec/composite/DocValuesProvider.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.index.codec.composite;
+
+import org.apache.lucene.codecs.DocValuesProducer;
+import org.apache.lucene.index.SortedNumericDocValues;
+
+import java.io.IOException;
+
+/**
+ * An interface that provides access to document values for a specific field.
+ *
+ * @opensearch.experimental
+ */
+public interface DocValuesProvider {
```

Comment:
This is quite confusing as we already have DocValuesProducer from Lucene. Lets rename.

Reply:
+1 to rename this. Also, can you add javadoc around how this new getSortedNumeric method adds value to the client impl?

Reply:
done!